src/libraries/Microsoft.PowerFx.Interpreter/Functions/LibraryUntypedObject.cs

```diff
@@ -423,7 +423,7 @@ public static FormulaValue CountRows_UO(IRContext irContext, UntypedObjectValue[
         {
             var impl = args[0].Impl;
 
-            if (impl.Type is ExternalType externalType && externalType.Kind == ExternalTypeKind.Array)
+            if (impl.Type is ExternalType externalType && (externalType.Kind == ExternalTypeKind.Array || externalType.Kind == ExternalTypeKind.ArrayAndObject))
```

Comment:
What does it mean for the ExternalTypeKind to be "ArrayAndObject"? Is it guaranteed to have an array length?